doc/source/whatsnew/v0.23.0.txt

```diff
@@ -418,6 +418,8 @@ I/O
 - Bug in :func:`read_sas` where a file with 0 variables gave an ``AttributeError`` incorrectly. Now it gives an ``EmptyDataError`` (:issue:`18184`)
 - Bug in :func:`DataFrame.to_latex()` where pairs of braces meant to serve as invisible placeholders were escaped (:issue:`18667`)
 - Bug in :func:`read_json` where large numeric values were causing an ``OverflowError`` (:issue:`18842`)
+- Type ``Interval`` now supported in :func:`DataFrame.to_excel` for all Excel file types (:issue:`19242`)
```

Comment:
use ``:class:`Interval` ``, and ``:class`Timdelta` `` (this is more user friendly as you almost always have a ``Timedelta``)